lib/signs/bus.ex

```diff
@@ -492,6 +536,102 @@ defmodule Signs.Bus do
     {messages, audios, tts_audios}
   end
 
+  @spec headway_message_for_configs([map()], [String.t()], DateTime.t()) ::
+          {:headway, [Message.Headway.t()]}
+  defp headway_message_for_configs(configs, stop_ids, current_time) do
+    configs
+    |> Enum.find(fn config ->
+      config[:headway_group] && config[:headway_direction_name]
+    end)
+    |> case do
+      %{headway_group: group, headway_direction_name: name} ->
+        message =
+          Headways.headway_message_sl(
+            group,
+            silver_line_destination(name),
+            stop_ids,
+            current_time
+          )
+
+        case message do
+          nil -> {:headway, []}
```

Comment:
Instead of generating an empty message here and then having to handle that downstream, what about reframing things a bit? If we built this logic into `configs_content`, then we wouldn't have to call it repeatedly, and we could simply not produce a message at all if we don't have the relevant headway data.